src/tests/ftest/server/daos_server_restart.py

```diff
@@ -0,0 +1,192 @@
+        and reformat.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DaosServerTest object."""
+        super(DaosServerTest, self).__init__(*args, **kwargs)
+
+    @fail_on(ServerFailed)
+    @fail_on(CommandFailure)
+    def restart_daos_server(self, reformat=True):
+        """method to perform server stop and start.
+        Args:
+            reformat (bool): always reformat storage, could be destructive.
+
+        """
+        self.log.info("=Restart daos_server, server stop().")
+        self.server_managers[0].stop()
+        self.log.info("=Restart daos_server, prepare().")
+        self.server_managers[0].prepare()
+        self.log.info("=Restart daos_server, detect_format_ready().")
+        self.server_managers[0].detect_format_ready()
+        self.log.info("=Restart daos_server, dmg storage_format.")
+        self.server_managers[0].dmg.exit_status_exception = False
+        self.server_managers[0].dmg.storage_format(reformat)
+        self.log.info("=Restart daos_server, detect_io_server_start().")
+        self.server_managers[0].detect_io_server_start()
+
+    @fail_on(ServerFailed)
+    @fail_on(CommandFailure)
+    def restart_daos_io_server(self, force=True):
+        """method to perform io_server stop and start by dmg.
+        Args:
+            force (bool): Force to stop the daos io_server.
+            Defaults to True.
+
+        """
+        self.server_managers[0].dmg.system_stop(force)
+        self.server_managers[0].dmg.system_start()
+
+    def get_pool_list(self):
+        """method to get the pool list contents"""
+        pool_list = self.get_dmg_command().get_output("pool_list")
+        self.log.info("get_pool-list: %s", pool_list)
+        return pool_list
+
+    def verify_pool_list(self, expected_pool_list=None):
+        """method to verify the pool list"""
+        if expected_pool_list is None:
+            expected_pool_list = []
+        pool_list = self.get_pool_list()
+        self.log.info(
+            "\n===Current pool-list:  %s\n===Expected pool-list: %s\n",
+            pool_list, expected_pool_list)
+        self.assertEqual(
+            pool_list, expected_pool_list,
+            "##Current pool-list mismatch with the expected pool-list.")
+
+    def create_pool_and_container(self):
+        """method to create pool and container"""
+        scm_size = self.params.get("scm_size", "/run/server/*/", 138000000)
+        num_of_pool = self.params.get("num_of_pool", "/run/server/*/", 3)
+        container_per_pool = self.params.get(
+            "container_per_pool", "/run/server/*/", 2)
+        for _ in range(num_of_pool):
+            dmg = self.get_dmg_command()
+            result = dmg.pool_create(scm_size)
+            uuid = result['uuid']
+            svc = result['svc']
+            daos_cmd = DaosCommand(self.bin)
+            daos_cmd.exit_status_exception = False
```

Comment:
I just noticed this line, why are you disabling exceptions for the DaosCommand object here? This means that if the command fails the test will continue without reporting the failure.

Reply:
Please see the new commit.